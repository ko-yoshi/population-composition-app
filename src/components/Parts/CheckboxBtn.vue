<template>
  <div class="checkbox-btn">
    <label :class="{ checked: model }">
      <input v-model="model" type="checkbox" @change="emits('change', model)" />
      {{ props.label }}
    </label>
  </div>
</template>
<script lang="ts">
  type Props = { label: string; checked?: boolean; bgColor?: string }
  type Emits = { (e: 'change', val: boolean): void }
</script>
<script setup lang="ts">
  const props = withDefaults(defineProps<Props>(), {
    label: '',
    checked: false,
    bgColor: '4287f5',
  })
  const emits = defineEmits<Emits>()
  const model = ref(props.checked)
</script>
<style lang="scss" scoped>
  .checkbox-btn {
    position: relative;
    display: inline-block;

    input {
      position: absolute;
      left: 0;
      opacity: 0;
    }

    label {
      display: inline-block;
      min-width: 6em;
      padding: 0.6em;
      font-size: 12px;
      text-align: center;
      cursor: pointer;
      background-color: #eee;
      border-radius: 5px;
      transition: 0.2s;

      &.checked {
        color: #fff;
        background-color: v-bind("'' + props.bgColor");
      }
    }
  }
</style>
