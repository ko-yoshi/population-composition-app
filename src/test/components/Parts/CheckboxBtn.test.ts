import { mount } from '@vue/test-utils'
import { expect, test, describe } from 'vitest'
import CheckboxBtn from '~/components/Parts/CheckboxBtn.vue'

describe('PartsCheckboxBtn', () => {
  const wrapper = mount(CheckboxBtn, {
    props: {
      label: 'テスト',
    },
  })
  const checkbox = wrapper.find('input[type="checkbox"]')
  const label = wrapper.find('label')

  test('props確認 labelのみ設定', () => {
    expect(wrapper.props()).toEqual({ label: 'テスト', checked: false, bgColor: '#4287f5' })
    expect(label.text()).toEqual('テスト')
  })

  test('props確認 全て設定', async () => {
    await wrapper.setProps({
      label: 'テスト2',
      checked: true,
      bgColor: '#123456',
    })
    expect(wrapper.props()).toEqual({ label: 'テスト2', checked: true, bgColor: '#123456' })
  })

  test('chekcedでlabelのclassにcheckedがつくか確認', async () => {
    await checkbox.setValue(true)
    expect(label.attributes('class')).toBe('checked')
  })

  test('emit確認', async () => {
    const emit = wrapper.emitted()
    await checkbox.setValue(true)
    expect(emit.change).toBeTruthy()
    expect(emit.change[0]).toEqual([true])
    await checkbox.setValue(false)
    expect(emit.change).toBeTruthy()
    expect(emit.change[1]).toEqual([false])
  })

  test('snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
