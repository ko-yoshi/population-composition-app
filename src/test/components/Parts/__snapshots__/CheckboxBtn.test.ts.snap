// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PartsCheckboxBtn > snapshot 1`] = `"<div data-v-6e17ff53=\\"\\" class=\\"checkbox-btn\\"><label data-v-6e17ff53=\\"\\" class=\\"\\"><input data-v-6e17ff53=\\"\\" type=\\"checkbox\\"> テスト2</label></div>"`;
